core/2d/CCAutoPolygon.cpp

```diff
@@ -616,38 +625,24 @@ TrianglesCommand::Triangles AutoPolygon::triangulate(const std::vector<Vec2>& po
             if (found)
             {
                 // if we found the same vertex, don't add to verts, but use the same vertex with indices
-                indices.emplace_back(j);
-                idx++;
+                indicesBuf[idx++] = j;
             }
             else
             {
                 // vert does not exist yet, so we need to create a new one,
                 auto c4b         = Color4B::WHITE;
                 auto t2f         = Tex2F(0, 0);  // don't worry about tex coords now, we calculate that later
-                V3F_C4B_T2F vert = {v3, c4b, t2f};
-                verts.emplace_back(vert);
-                indices.emplace_back(vdx);
-                idx++;
-                vdx++;
+
+                indicesBuf[idx++] = vdx++;
+
+                vertsBuf.emplace(V3F_C4B_T2F{v3, c4b, t2f});
             }
         }
     }
-    for (auto&& j : p2points)
-    {
-        delete j;
-    };
-
-    // now that we know the size of verts and indices we can create the buffers
-    V3F_C4B_T2F* vertsBuf = new V3F_C4B_T2F[verts.size()];
-    memcpy(vertsBuf, verts.data(), verts.size() * sizeof(V3F_C4B_T2F));
-
-    unsigned short* indicesBuf = new unsigned short[indices.size()];
-    memcpy(indicesBuf, indices.data(), indices.size() * sizeof(short));
 
-    // Triangles should really use std::vector and not arrays for verts and indices.
-    // Then the above memcpy would not be necessary
-    TrianglesCommand::Triangles triangles = {vertsBuf, indicesBuf, (unsigned int)verts.size(),
-                                             (unsigned int)indices.size()};
+    // Good concept: the pod_vector support release internal buffer ownership
+    TrianglesCommand::Triangles triangles = {vertsBuf.release_pointer(), indicesBuf, (unsigned int)vdx,
+                                             (unsigned int)idx};
```

Comment:
Please let's do what is written here

> Triangles should really use std::vector and not arrays for verts and indices

I am afraid that if we will introduce pod_vector then we will find many places in the code where it can be used. 
But the proper way here is to remove c style array and use std::vector everywhere

Reply:
OK, will remove pod_vector use at here